src/plugins/dataflow/nodes/dataflow-node-plot.tsx

```diff
@@ -30,30 +30,27 @@ enum Zoom {
   Out
 }
 
+//This will not behave as expected for tickMin > 0 (or if both values are negative, but that probably won't happen).
+// Not sure what should happen in that case. Part of what @tealefristoe was recommending was an algorithm that worked
+//independent of the values involved.
+
 export const DataflowNodePlot: React.FC<INodePlotProps> = (props) => {
   if (!props.display) return null;
 
   const handleClickOffset = (zoomDir: Zoom) => {
     const max = maxY(props.data);
     const min = minY(props.data);
     const difference = Math.abs(max - min);
-    const offset = 0.1 * difference;
-
-    if (zoomDir === Zoom.In ){
-      props.data.data.tickMax = max - offset;
-      props.data.data.tickMin = min + offset;
-    }
-
-    if (zoomDir === Zoom.Out){
-      props.data.data.tickMax = max + offset;
-      props.data.data.tickMin = min - offset;
-    }
+    const midpoint = (max + min)/2;
```

Comment:
One more thing to consider: by using `max` and `min` here, the center will be fixed as soon as the user pushes the + or - button and will never change. If you use `props.data.data.dsMax` and `dsMin` here, the midpoint will shift to the current center of the data, rather than the center of the data when the user first adjusted the scale.

I'm not sure what the preferred behavior would be. I feel like keeping it centered on the current data would be preferable, but that's a project team/design call. And even if we want to keep the graph centered on the data's current center, this might not be the place to make that happen, since it would only change when the user pushes + or - here. So I guess I'd suggest keeping this as is for now, but keep in mind that this behavior may need to change with feedback from Leslie or others.

Reply:
Yeah I wouldn't mess with this right now. We've already held this up long enough and again the case that the project currently cares about is really about increasing the top of the y axis. The current behavior works ok for the EMG signals. I was able to recreate the weird bug with lots of extra trailing digits but I will report that separately. We need to get this branch merged.